sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -485,6 +491,11 @@ relevanceField
     | stringLiteral
     ;
 
+nestedField
+    : qualifiedName
+    | stringLiteral (DOT stringLiteral)*
```

Comment:
@dai-chen If we go in this direction we would have to merge to 3.x since this will be a breaking change. This also means that legacy will have to be deprecated in 3.x instead of 2.x. 
Another option would be to support string arguments in 2.x in order to deprecate in 2.x. Then we can push a change for 3.x that will change the behaviour.

What do you think?